app/static/js/graphs/bubble-graph.js

```diff
@@ -143,15 +238,110 @@ function drawBubbleGraph(selector, data, replication) {
             }
         });
 
-        parentSVG.addClass("term-" + selected.attr('value'));
+        parentSVG.addClass("term-" + selected_.attr('value'));
 
         var filters = parentSVG.attr("class");
-        reDrawBubbleGraph(data, filters, selector, yScale, sizeScale, tickMax);
+
+        reDrawBubbleGraph(data, filters, selector, xScale, yScale, sizeScale, tickMax); //Add xScale
+
+        //redraw the selected data

```

Comment:
Tested, looks OK to me

Reply:
I think I'm using the same syntax as the redraw code above.. the process is a bit complex since the original `reDrawBubbleGraph` function is not complete in terms of dealing with selections (such as the 3 characters selector)... In the original function, the graph is completely redrawn and the selected points were redrawn again (to make the selected points (not opaque) cover the opaque points), thus, there are additional data points been added into the dataset and lead confusion in further plots.

so I add 2 additional traits to every data point ("PatternSelection" and "ancestrySelection") in the `reDrawBubbleGraph` function and use them to judge whether these points should be deleted after invoking the function.  so the `var selectedBefore` (been mentioned in last question)is actually trying to capture the additional points and then delete them using `$(selectedBefore_pattern).remove();`

Reply:
Resolving this one but see other comments.